net/instaweb/http/async_fetch.cc

```diff
@@ -276,6 +276,34 @@ void SharedAsyncFetch::HandleHeadersComplete() {
   base_fetch_->HeadersComplete();
 }
 
+OutputSanitizingAsyncFetch::OutputSanitizingAsyncFetch(AsyncFetch* base_fetch)
+    : SharedAsyncFetch(base_fetch) {
+  DCHECK(base_fetch != nullptr);
+}
+
+OutputSanitizingAsyncFetch::~OutputSanitizingAsyncFetch() {
+}
+
+bool OutputSanitizingAsyncFetch::SanitizeResponseHeaders() {
+  if (response_headers() != nullptr &&
+      response_headers()->RemoveAllWithPrefix("@")) {
+    response_headers()->ComputeCaching();
+    return true;
+  }
+  return false;
+}
+
+void OutputSanitizingAsyncFetch::HandleHeadersComplete() {
+  SanitizeResponseHeaders();
+  SharedAsyncFetch::HandleHeadersComplete();
+}
+
+void OutputSanitizingAsyncFetch::HandleDone(bool success) {
+  SanitizeResponseHeaders();
```

Comment:
This one might be dubious threading-wise? Though I think our story on this hasn't been very clean.

Reply:
Welll... There is some documentation here:
https://github.com/pagespeed/mod_pagespeed/blob/master/net/instaweb/http/public/async_fetch.h#L114
.... which is not a very good place to have it. 

At any rate thread issues seem very likely for this check if you look at how the code in RemoveAllWithPrefix looks like.